### PR TITLE
save user to local storage task#85

### PR DESCRIPTION
**Pull request by SlakvichAleksandr**

### CODE REVIEW

**Comment by SlakvichAleksandr** [comment]
#85 - таска в трелло

File: frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts [hunk 1]
```diff
@@ -12,9 +11,9 @@ export class ExternalBookingPageComponent {
...
 
     // eslint-disable-next-line no-empty-function
-    constructor(public spinnerService: SpinnerService, private userService: UserService) {
-        this.userService.getCurrentUser().subscribe((user) => {
-            this.selectedUser = user;
-        });
+    constructor(public spinnerService: SpinnerService) {
+        const user = localStorage.getItem('user');
+
+        this.selectedUser = JSON.parse(user!) as IUser;
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts]: You cannot use local storage in other services

File: frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts [hunk 2]
```diff
@@ -112,7 +108,9 @@ export class UserProfilePageComponent extends BaseComponent implements OnInit {
             .editUser(editedUser)
             .pipe(this.untilThis)
             .subscribe(
-                () => {
+                (user) => {
+                    localStorage.removeItem('user');
+                    localStorage.setItem('user', JSON.stringify(user));
```
Comment (tatianahutii) [inline frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts]: See the previous comment

File: frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts [hunk 3]
```diff
@@ -74,24 +73,21 @@ export class UserProfilePageComponent extends BaseComponent implements OnInit {
...
-                    dateFormat: user.dateFormat,
-                    timeFormat: user.timeFormat,
-                    language: user.language,
-                    timeZone: user.timeZone,
-                    image: user.image,
-                });
-                this.imageUrl = user.image;
-                this.changeCountryCode(this.userForm);
-            });
+        const currentUser = localStorage.getItem('user');
```
Comment (tatianahutii) [inline frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts]: See the previous comment

File: frontend/src/app/modules/availability/availability-page/availability-page.component.ts [hunk 4]
```diff
@@ -40,15 +35,10 @@ export class AvailabilityPageComponent extends BaseComponent {
...
-            .pipe(this.untilThis)
-            .subscribe((resp) => {
-                if (resp) {
-                    this.currentUser = resp;
-                    this.getUserPersonalAndTeamSlots();
-                }
-            });
+        const user = localStorage.getItem('user');
+
+        this.currentUser = JSON.parse(user!) as IUser;
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/availability-page/availability-page.component.ts]: See the previous comment

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts [hunk 5]
```diff
@@ -72,7 +72,8 @@ export class SignUpFormComponent extends BaseComponent {
                 })
                 .pipe(this.untilThis)
                 .subscribe(
-                    () => {
+                    (user) => {
+                        localStorage.setItem('user', JSON.stringify(user));
```
Comment (tatianahutii) [inline frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: Move this to the Auth method

File: frontend/src/app/core/services/user.service.ts [hunk 6]
```diff
@@ -14,7 +15,13 @@ export class UserService {
...
 
     public getCurrentUser() {
-        return this.httpService.getRequest<IUser>(`${this.routePrefix}/current`);
+        return this.httpService.getRequest<IUser>(`${this.routePrefix}/current`).pipe(
+            map((resp) => {
+                this.setUser(resp);
+
+                return resp;
+            }),
+        );
```
Comment (tatianahutii) [inline frontend/src/app/core/services/user.service.ts:24]: We need not all this data

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts [hunk 7]
```diff
@@ -9,12 +9,10 @@ import { UserService } from '@core/services/user.service';
     styleUrls: ['./external-booking-page.component.sass'],
 })
 export class ExternalBookingPageComponent {
-    public selectedUser: IUser;
+    public selectedUser: ILocalUser;
 
     // eslint-disable-next-line no-empty-function
     constructor(public spinnerService: SpinnerService, private userService: UserService) {
```
Comment (tatianahutii) [inline frontend/src/app/modules/external-booking/external-booking-page/external-booking-page.component.ts:14]: please delete eslint-disable

File: frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts [hunk 8]
```diff
@@ -110,7 +109,9 @@ export class UserProfilePageComponent extends BaseComponent implements OnInit {
             .editUser(editedUser)
             .pipe(this.untilThis)
             .subscribe(
-                () => {
+                (user) => {
+                    this.userService.removeUser();
+                    this.userService.setUser(user);
```
Comment (tatianahutii) [inline frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts]: Please create a separate method in the service for updating

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**